packages/graphql/src/schema-model/generate-model.ts

```diff
@@ -153,15 +239,27 @@ function getLabels(definition: ObjectTypeDefinitionNode, nodeDirectiveArguments:
 
 function generateField(field: FieldDefinitionNode): Attribute | undefined {
     const typeMeta = getFieldTypeMeta(field.type); // TODO: without originalType
-    if (SCALAR_TYPES.includes(typeMeta.name)) {
+    if (isAttributeType(typeMeta.name)) {
         const annotations = createFieldAnnotations(field.directives || []);
         return new Attribute({
             name: field.name.value,
             annotations,
+            type: typeMeta.name,
+            isArray: Boolean(typeMeta.array),
         });
     }
 }
 
+function isAttributeType(typeName: string): typeName is AttributeType {
```

Comment:
Potentially we can start moving these utility functions in some parser utils file.